cumulusci/core/config/marketing_cloud_service_config.py

```diff
@@ -0,0 +1,29 @@
+from cumulusci.core.config import ServiceConfig
+from cumulusci.oauth.client import OAuth2Client
+from cumulusci.oauth.client_info import OAuthClientInfo
+
+
+class MarketingCloudServiceConfig(ServiceConfig):
+    def __init__(self, config, keychain):
+        super().__init__(config=config)
+        self._client_info = keychain.get_service("oauth-client", config["oauth_client"])
+
+    def connect(client_info: OAuthClientInfo):
```

Comment:
As I mentioned in org_connect, I think this connect method should receive a dict with all the service attributes the user was prompted for, rather than an OAuthClientInfo specifically. It's more flexible.